packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -515,159 +524,169 @@ export default class TableResize implements EditorPlugin {
         this.editor.runAsync(() => this.resizeCells(e));
     };
 
-    private resizeCells = (e: MouseEvent) => {
-        this.setTableResizer(null);
-        if (this.resizingState === ResizeState.None) {
-            return;
-        } else if (this.resizingState === ResizeState.Both) {
-            let rect = normalizeRect(this.currentTable.getBoundingClientRect());
-            let vtable = new VTable(this.currentTable);
+    private resizeTable = (mouseX: number, mouseY: number) => {
+        this.resizingVtable.preProcessTable();
+        if (!this.isReadyForResizingTable) {
+            const rect = this.resizingVtable.table.getBoundingClientRect();
+            this.currentTableVerticalBorder = this.isRTL ? rect.left : rect.right;
+            this.currentTableHorizontalBorder = rect.bottom;
+
+            for (let i = 0; i < this.resizingVtable.cells.length; i++) {
+                this.cellsBeforeResize.push([]);
+                for (let j = 0; j < this.resizingVtable.cells[i].length; j++) {
+                    this.cellsBeforeResize[i].push(
+                        this.resizingVtable.cells[i][j].td?.cloneNode(true) as HTMLTableCellElement
+                    );
+                }
+            }
```

Comment:
Actually what you need is just original width and height, why do you need to clone the whole TD?

And since we never preprocess a table again during resizing, you can just use vtable.cells[][].width/height as original size

Reply:
And no need to have isReadyForResizingTable.

Reply:
Fixed the first one, but I think we need isReadyForResizingTable to calculate the ratioX/ratioY based on the original table size (not the original td size.), right? The table size is being changed during resizing, so we need isReadyForResizingTable  so that we calculate only once.

Reply:
fixed